### PR TITLE
Update sink rate limit fields

### PR DESCRIPTION
**Pull request by jpggvilaca**
### Summary

* Fix last bug of https://github.com/open-edge-platform/training_extensions/issues/5532
* Add rate limit fields as requested

Screenshot 2026-03-04 at 15 23 19
Screenshot 2026-03-04 at 15 41 03
Screenshot 2026-03-04 at 15 41 08
Screenshot 2026-03-04 at 15 41 11
Screenshot 2026-03-04 at 15 42 45

### Checklist

- [ ] The PR title and description are clear and descriptive
- [ ] I have manually tested the changes
- [ ] All changes are covered by automated tests
- [ ] All related issues are linked to this PR (if applicable)
- [ ] Documentation has been updated (if applicable)

### CODE REVIEW

File: application/ui/src/features/inference/sinks/utils.test.ts [hunk 1]
```diff
@@ -46,3 +46,83 @@ describe('getObjectFromFormData', () => {
...
+        expect(formatRateLimit(null)).toBe('Not set');
+        expect(formatRateLimit(0)).toBe('Not set');
+    });
+
+    it('formats singular sample and second labels', () => {
+        expect(formatRateLimit(1)).toBe('1 sample every 1 second');
+    });
+
+    it('formats plural samples for rates above one', () => {
+        expect(formatRateLimit(2)).toBe('2 samples every 1 second');
```
Comment (dwesolow) [inline application/ui/src/features/inference/sinks/utils.test.ts:135]: what about `1 sample every 0.5 second`? it will also rate limit 2.

File: application/ui/src/features/inference/sinks/utils.ts [hunk 2]
```diff
@@ -48,3 +74,45 @@ export const getObjectFromFormData = (keys: FormDataEntryValue[], values: FormDa
...
+
+    if (samplesValue === null || secondsValue === null) {
+        return null;
+    }
+
+    const samplesRaw = toStringAndTrim(samplesValue);
+    const secondsRaw = toStringAndTrim(secondsValue);
+
+    const samples = parseFiniteNumber(samplesRaw);
+    const seconds = parseFiniteNumber(secondsRaw);
```
Comment (dwesolow) [inline application/ui/src/features/inference/sinks/utils.ts]: why do we need these utils? I thought that the value is a string that includes a number so only `parseFloat` would suffice, no?

File: application/ui/src/features/inference/sinks/utils.ts [hunk 3]
```diff
@@ -48,3 +74,45 @@ export const getObjectFromFormData = (keys: FormDataEntryValue[], values: FormDa
...
+        const normalizedSeconds = toDisplayNumber(seconds);
+        const secondsLabel = normalizedSeconds === '1' ? 'second' : 'seconds';
+
+        return `1 sample every ${normalizedSeconds} ${secondsLabel}`;
+    }
+
+    const normalizedSamples = toDisplayNumber(normalizedRateLimit);
+    const sampleLabel = normalizedSamples === '1' ? 'sample' : 'samples';
+
+    return `${normalizedSamples} ${sampleLabel} every 1 second`;
```
Comment (dwesolow) [inline application/ui/src/features/inference/sinks/utils.ts:92]: I think it won't show correct initial values, e.g `0.5` can be  1/2 or 5/10 etc. Could we store both values on the server side and then server can do `rate_limit = rate_sample / rate_seconds`?

File: application/ui/src/features/inference/sinks/utils.test.ts (hunk 1 above)
Comment (jpggvilaca) [reply application/ui/src/features/inference/sinks/utils.test.ts:135]: Correct, makes sense?

File: application/ui/src/features/inference/sinks/utils.ts (hunk 2 above)
Comment (jpggvilaca) [reply application/ui/src/features/inference/sinks/utils.ts]: But the string can be very weird like `1.2,3` or use just commas or just spaces, or contain special characters.

File: application/ui/src/features/inference/sinks/utils.ts (hunk 3 above)
Comment (jpggvilaca) [reply application/ui/src/features/inference/sinks/utils.ts:92]: I think asking to change the API will take longer. What is important here is the rate limit, not which values you used to get there. Makes sense?

Comment (dwesolow) [reply application/ui/src/features/inference/sinks/utils.ts:92]: maybe, but we will deceive the user, if they set 16 samples / 4 seconds and we will show 4 samples / 1 second, it might not be true?

File: application/ui/src/features/inference/sinks/utils.ts (hunk 2 above)
Comment (dwesolow) [reply application/ui/src/features/inference/sinks/utils.ts]: Why it can be like this? It's number field so it should not happen?

File: application/ui/src/features/inference/sinks/utils.test.ts (hunk 1 above)
Comment (dwesolow) [reply application/ui/src/features/inference/sinks/utils.test.ts:135]: I meant https://github.com/open-edge-platform/training_extensions/pull/5682#discussion_r2889338958, maybe we should ask BE team if we should care about this.

File: application/ui/src/features/inference/sinks/utils.ts [hunk 4]
```diff
@@ -39,6 +39,32 @@ export enum WebhookHttpMethod {
...
+    }
+
+    const normalizedValue = compactValue.replace(',', '.');
+    const parsedValue = Number(normalizedValue);
+
+    return Number.isFinite(parsedValue) ? parsedValue : null;
+};
+
+const toDisplayNumber = (value: number) => {
+    const roundedValue = Number(value.toFixed(2));
```
Comment (camiloHimura) [inline application/ui/src/features/inference/sinks/utils.ts]: `Math.round` is more descriptive than using toFixed + Casting

File: application/ui/src/features/inference/sinks/utils.test.ts [hunk 5]
```diff
@@ -46,3 +46,83 @@ describe('getObjectFromFormData', () => {
...
+    });
+
+    it('returns null for invalid mixed separators', () => {
+        const formData = new FormData();
+        formData.set('rate_limit_samples', '1,2.3');
+        formData.set('rate_limit_seconds', '1');
+
+        expect(rateLimitFromFormData(formData)).toBeNull();
+    });
+});
```
Comment (camiloHimura) [inline application/ui/src/features/inference/sinks/utils.test.ts:120]: It might need some test cases with empty strings and `NaN`. wdyt?

File: application/ui/src/features/inference/sinks/rate-limit/rate-limit-fields.component.tsx [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    rateLimit: number | null | undefined;
+};
+
+export const RateLimitFields = ({ rateLimit }: RateLimitFieldsProps) => {
+    const samples = positiveNumberOrUndefined(rateLimit);
+    const seconds = samples !== undefined ? 1 : undefined;
+
+    return (
+        <Flex direction='row' gap='size-100' alignItems={'end'}>
+            <NumberField label='Samples' name='rate_limit_samples' minValue={0.1} step={0.1} defaultValue={samples} />
```
Comment (camiloHimura) [inline application/ui/src/features/inference/sinks/rate-limit/rate-limit-fields.component.tsx:18]: Should they be required?

File: application/ui/src/features/inference/sinks/utils.ts (hunk 4 above)
Comment (jpggvilaca) [reply application/ui/src/features/inference/sinks/utils.ts]: We want the 2 decimals points, with round you wont get that. But i did remove the unnecessary `Number`

File: application/ui/src/features/inference/sinks/utils.ts (hunk 3 above)
Comment (jpggvilaca) [reply application/ui/src/features/inference/sinks/utils.ts:92]: I see your point, but i would also argue that the user wouldnt do that. If you want half, why would you do higher numbers and combinations? you would always use the smallest combo possible. Anyway this is more an UX issue, ill ask on the next meeting

File: application/ui/src/features/inference/sinks/utils.ts (hunk 2 above)
Comment (jpggvilaca) [reply application/ui/src/features/inference/sinks/utils.ts]: "should" yeah :D https://react-spectrum.adobe.com/v3/NumberField.html

File: application/ui/src/features/inference/sinks/rate-limit/rate-limit-fields.component.tsx (hunk 6 above)
Comment (jpggvilaca) [reply application/ui/src/features/inference/sinks/rate-limit/rate-limit-fields.component.tsx:18]: @leoll2 ^

File: application/ui/src/features/inference/sinks/utils.test.ts (hunk 5 above)
Comment (jpggvilaca) [reply application/ui/src/features/inference/sinks/utils.test.ts:120]: Sounds good! added

File: application/ui/src/features/inference/sinks/utils.test.ts (hunk 1 above)
Comment (jpggvilaca) [reply application/ui/src/features/inference/sinks/utils.test.ts:135]: @leoll2  ^

File: application/ui/src/features/inference/sinks/utils.ts (hunk 4 above)
Comment (jpggvilaca) [reply application/ui/src/features/inference/sinks/utils.ts]: Actually scratch that. We dont need the decimals at all. Updated :)

File: application/ui/src/features/inference/sinks/utils.ts (hunk 3 above)
Comment (leoll2) [reply application/ui/src/features/inference/sinks/utils.ts:92]: I agree it's pointless to change the API for this. 16/4 and 4/1 are equivalent, just normalized differently. An alternative approach is to use a single input field, fixing the other value to 1, as in `"1 frame every [ ] seconds"`. If so, however, we should add an explicit toggle button to enable/disable the rate limiter, because it's not obvious that leaving the field blank disables the limiter.

File: application/ui/src/features/inference/sinks/rate-limit/rate-limit-fields.component.tsx (hunk 6 above)
Comment (jpggvilaca) [reply application/ui/src/features/inference/sinks/rate-limit/rate-limit-fields.component.tsx:18]: `No, it is an optional setting. If rate_limit=null, the server doesn't enforce a limit`

File: application/ui/src/features/inference/sinks/utils.ts (hunk 2 above)
Comment (jpggvilaca) [reply application/ui/src/features/inference/sinks/utils.ts]: Checked props and removed `parseFiniteNumber`, check last commit

Comment (dwesolow) [reply application/ui/src/features/inference/sinks/utils.ts]: ah, it's American notation, imo we should not care about this

File: application/ui/src/features/inference/sinks/utils.test.ts (hunk 1 above)
Comment (leoll2) [reply application/ui/src/features/inference/sinks/utils.test.ts:135]: "2 samples every 1 second" and "1 sample every 0.5 second" are both acceptable for me

File: application/ui/tests/inference/inference.spec.ts [hunk 7]
```diff
@@ -314,7 +314,8 @@ test('Inference', async ({ streamPage, page, network }) => {
         await page.getByRole('button', { name: 'Add new sink' }).click();
         await page.getByRole('button', { name: 'Folder' }).click();
         await page.locator('input[name="name"]').fill('New Folder');
-        await page.locator('input[aria-roledescription="Number field"]').fill('5');
+        await page.locator('input[aria-roledescription="Number field"]').first().fill('5');
```
Comment (dwesolow) [inline application/ui/tests/inference/inference.spec.ts:317]: np. we usually try to use `getByX` whenever we can, e.g. `page.getByRole('textbox', { name: 'Number field'} )` (didn't check if this work, just pseudo code)

File: application/ui/src/features/inference/sinks/utils.ts (hunk 3 above)
Comment (dwesolow) [reply application/ui/src/features/inference/sinks/utils.ts:92]: @jpggvilaca do you want to add toggle button in this pr?

**Review by dwesolow — APPROVED** [review]

File: application/ui/src/features/inference/sinks/utils.ts (hunk 3 above)
Comment (jpggvilaca) [reply application/ui/src/features/inference/sinks/utils.ts:92]: Discussed offline. No need for this
